cmd/crc-embedder/cmd/embed.go

```diff
@@ -124,9 +143,25 @@ func getEmbedFileList(goos string, destDir string) []string {
 	return fileList
 }
 
-func downloadDataFiles(goos string, destDir string) ([]string, error) {
+func downloadDataFiles(goos string, components []string, destDir string) ([]string, error) {
 	downloadedFiles := []string{}
 	downloads := dataFileUrls[goos]
+
+	if len(components) > 0 {
+		for _, component := range components {
+			if dl, ok := downloads[component]; ok {
+				filename, err := download.Download(dl.url, destDir, dl.permissions, nil)
+				if err != nil {
+					return nil, err
+				}
+				downloadedFiles = append(downloadedFiles, filename)
+			} else {
+				logging.Warnf("'%s' is not available, ensure goos flag is used. Supported component names are: %s", component, strings.Join(componentNames, ", "))
+			}
+		}
+		return downloadedFiles, nil
```

Comment:
I'd prefer if we had a single loop for both `len(components) == 0` and `len(components != 0`.
This could take the form of:
```
for component, dl := range downloads {
    if !shouldDownload(components, component) {
        continue
    }
    downloadedFiles = append(downloadedFiles, filename)
}
```
If you want to keep some error reporting for bad component names, you can either check this before running downloadDataFiles, or you can remove the downloaded components from the `components` map, and check if it's empty or not at the end of the loop.

Reply:
i thought about this then preferred two loops to avoid an `if` check for the component name on every iteration, will update!